### ISSUE TITLE
latency when directiong audio to stdout

### ISSUE THREAD

**Issue by vinaynk** [stsaz/fmedia#25]
Hi folks,

First of all, great thanks to all who are involved in the development of fmedia. It really reduces the pain of having to deal with different kinds of audio sources across multiple platforms.

I am currently trying to capture audio from mic using the option "--out=@stdout.wav". It works good, except that the latency is a bit high. It seems the output is only dumped every 2 sec or so. Is there a way to set this small (like 100ms) ?

Best Regards
.

**Comment by stsaz** [comment]
Hi!
I'm glad you like fmedia!

First of all you should set `buffer_size` (in fmedia.conf) smaller than 64k:
```
mod_conf "#file.out" {
	buffer_size 64k
```
This setting directly affects how often "stdout" module flushes its data.
*Note that this setting stopped working in v1.11, so you should use v1.10.1 instead.* - I'll fix it in the next version.

Second, in fmedia.conf find `AUDIO I/O` section and set `buffer_length` to a smaller value:
e.g.:
```
mod_conf "wasapi.in" {
	...
	# in msec
	buffer_length 100
```
On Windows it's `wasapi.in`
On Linux it's `alsa.in`
*Note that hardware may not support the value which is too small!*

**Comment by stsaz** [comment]
> Note that this setting stopped working in v1.11, so you should use v1.10.1 instead. - I'll fix it in the next version.

fixed in v1.12
